ferry.go

```diff
@@ -452,11 +473,14 @@ func (f *Ferry) Initialize() (err error) {
 
 	// The iterative verifier needs the binlog streamer so this has to be first.
 	// Eventually this can be moved below the verifier initialization.
-	f.BinlogStreamer = f.NewBinlogStreamer(f.SourceDB, f.Config.Source)
+	f.BinlogStreamer = f.NewSourceBinlogStreamer()
 
 	if !f.Config.SkipTargetVerification {
-		targetBinlogStreamer := f.NewBinlogStreamer(f.TargetDB, f.Config.Target)
-		targetBinlogStreamer.LogTag = "tgt_binlog_streamer"
+		targetBinlogStreamer, err := f.NewTargetBinlogStreamer()
```

Comment:
This is fine, but it was a bit surprising at first to see the the `SourceBinlogStreamer` doesn't return an error while this one does. Makes sense, but it did cause me to re-read that code again.

Reply:
Right. Would you prefer to see a common interface (ie: both return errors) even if the `NewSourceBinlogStreamer` will always return a `nil` error for now?

Reply:
I think what you have here is fine so we can leave as is. I just tend to write a comment when I spend more than 5 minutes investigating a single line of code, which maybe a bad rule to follow as in this case it didn't result in anything.